### ISSUE TITLE
Contributing docs for JetBrains Rider users

### ISSUE THREAD

**Issue by ShreyasJejurkar** [dotnet/aspnetcore#58722]
As we all know recently JetBrains Rider has now been made free for use in non-commercial environments (https://blog.jetbrains.com/dotnet/2024/10/16/rider-reveal-livestream-big-news-for-dotnet-and-game-devs/), I guess many of the open source developers will use it because of the speed and efficiency that it offers and also of course because of "Resharper". 
So I guess we should have a section for riders users as well in our contributing docs (docs/BuildFromSource.md) just like we have for "Visual Studio" users. I know VS is a product of MS, and it will be recommended but on the other side, JB rider is Superior for first-class .NET development as well. 

The thing is .NET open source developers using rider should not feel stuck when contributing to the aspnetcore repository. They should get the same support, documentation, and dev utilities (i.e. `startvs.cmd`) just like Visual Studio. 

I started exploring this path already, but I thought let me open issue first, to grab other people thoughts on this. 

Thanks, 
Shreyas

**Comment by xC0dex** [comment]
I agree, that would be amazing. I think in general it _should_ work. Rider detects the correct `dotnet.exe` and the SDK inside the repo. However, during the build I'm getting some errors like

```
`The imported file "$(VCTargetsPath)\Microsoft.Cpp.Default.props" does not exist and appears to be part of a Visual Studio component. This file may require MSBuild.exe in order to be imported successfully, and so may fail to build in the dotnet CLI.`
```

**Comment by xC0dex** [comment]
I’d like to give Rider users a quick update: It works quite well in Rider! If you follow the BuildFromSource.md guide for Visual Studio Code but run `rider .` instead of `code .`, it works. However, you need to use the ReSharper build. Otherwise, you get the error I mentioned above.
